org2jekyll.el

```diff
@@ -152,31 +172,52 @@ POST-TITLE is the title.
 POST-DESCRIPTION is the description.
 POST-TAGS is the tags
 POST-CATEGORIES is the categories."
-  (format org2jekyll-jekyll-org-post-template blog-layout blog-author post-date (org2jekyll--yaml-escape post-title) post-description post-tags post-categories))
+  (format org2jekyll-jekyll-org-post-template blog-type blog-layout blog-author post-date (org2jekyll--yaml-escape post-title) post-description post-tags post-categories))
 
 (defun org2jekyll--draft-filename (draft-dir title)
   "Compute the draft's filename from the DRAFT-DIR and TITLE."
   (concat draft-dir (org2jekyll--make-slug title) org2jekyll-jekyll-post-ext))
 
+(defsubst org2jekyll--get-post-types ()
+  "Return a list of post types.
+
+3 sources are queried, the result is uniqfied."
+  (let ((predefined-types '("page" "post"))
+        (prj-types (--keep (when (string= (plist-get (cdr it) :base-extension)
+                                          "org")
+                             (car it))
+                           org-publish-project-alist)))
+    (-uniq (append predefined-types prj-types org2jekyll-jekyll-draft-user-types))))
```

Comment:
Ok, so this generates the list of possible `type` (as you call them) which will be used as completion when preparing a draft.